src/hotspot/share/utilities/resourceHash.hpp

```diff
@@ -195,7 +217,51 @@ class ResourceHashtable : public ResourceObj {
       ++bucket;
     }
   }
+
+ protected:
+  void resize(unsigned new_size) {
+    Node** old_table = _table;
+    _table = alloc_table(new_size);
+
+    Node* const* bucket = old_table;
+    const unsigned old_size = size();
+    while (bucket < &old_table[old_size]) {
+      Node* node = *bucket;
+      while (node != NULL) {
+        Node* next = node->_next;
+        unsigned hash = HASH(node->_key);
+        unsigned index = hash % new_size;
+
+        node->_next = _table[index];
+        _table[index] = node;
+
+        node = next;
+      }
+      ++bucket;
+    }
+
+    FREE_C_HEAP_ARRAY(Node*, old_table);
+  }
 };
 
+template<
+    typename K, typename V,
+    // xlC does not compile this:
+    // http://stackoverflow.com/questions/8532961/template-argument-of-type-that-is-defined-by-inner-typedef-from-other-template-c
+    //typename ResourceHashtableFns<K>::hash_fn   HASH   = primitive_hash<K>,
+    //typename ResourceHashtableFns<K>::equals_fn EQUALS = primitive_equals<K>,
```

Comment:
Can you remove this xlC comment?  Not sure why we care.

Reply:
Done.